src/sdam/monitor.ts

```diff
@@ -363,8 +374,10 @@ function checkServer(monitor: Monitor, callback: Callback<Document | null>) {
       }
 
       monitor.connection = conn;
-      monitor.emit(
+      monitor.emitAndLogHeartbeat(
         Server.SERVER_HEARTBEAT_SUCCEEDED,
+        monitor[kServer].topology.s.id,
+        conn.id,
```

Comment:
We are passing in a mixture of driver-generated id and server connection id. This needs to also use the id that was provided on the handshake: `connection.hello.connectionId`